### PR TITLE
feat(origin): add fastify instance to origin function

### LINKED ISSUE #5: Pass fastify instance to origin function
Hi. In order to allow usage of previously registered modules and plugins (e.g. like fastify-mongo or others) in origin function it will be good if fastify instance will be passed there as parameter.

Now it's only an option to use closure for that, but that limits a bit where that function could be declared.

### PR DESCRIPTION
**Pull request by SkeLLLa**
~~BREAKING CHANGE: changed origin function parameters~~
Closes #5

Edit @delvedor:
Bind Fastify to origin function.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ function fastifyCors (fastify, opts, next) {
   function configureOrigin (req, reply, callback) {
     var reqOrigin = req.headers.origin
     if (isOriginFunction) {
-      var result = origin(reqOrigin, _onOrigin)
+      var result = origin(reqOrigin, fastify, _onOrigin)
```
Comment (delvedor) [inline index.js]: Why not bind the Fastify instance? Same result, no breaking change (and is also in line with what we do for hooks, middlewares...)

Comment (SkeLLLa) [reply index.js]: See https://github.com/fastify/fastify-cors/issues/5#issuecomment-429162101

Comment (mcollina) [reply index.js]: I'm +1 for binding.

File: README.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ You can use it as is without passing any option, or you can configure it as expl
...
   - `RegExp` - set `origin` to a regular expression pattern which will be used to test the request origin. If it's a match, the request origin will be reflected. For example the pattern `/example\.com$/` will reflect any request that is coming from an origin ending with "example.com".
   - `Array` - set `origin` to an array of valid origins. Each origin can be a `String` or a `RegExp`. For example `["http://example1.com", /\.example2\.com$/]` will accept any request from "http://example1.com" or from a subdomain of "example2.com".
-  - `Function` - set `origin` to a function implementing some custom logic. The function takes the request origin as the first parameter and a callback (which expects the signature `err [object], allow [bool]`) as the second, *async-await* and promises are supported as well.
+  - `Function` - set `origin` to a function implementing some custom logic. The function takes the request origin as the first parameter, `fastify` instance as a second and a callback (which expects the signature `err [object], allow [bool]`) as the last, *async-await* and promises are supported as well.
```
Comment (mcollina) [inline README.md]: I think this needs to be updated as well.

Comment (SkeLLLa) [reply README.md]: Sorry, that file wasn't in commit for some reason. Added it as well.

**Comment by cemremengu** [comment]
@mcollina ping

**Comment by cemremengu** [comment]
Thanks, will this be a patch?

**Comment by mcollina** [comment]
minor

Il giorno mar 23 ott 2018 alle 20:36 Cemre Mengu <notifications@github.com>
ha scritto:

**Pull request merged**